.github/workflows/DevEnv-frontend-BuildImageAndPush.yml

```diff
@@ -1,36 +1,102 @@
-# This is a basic workflow to help you get started with Actions
-
 name: DevEnv-frontend-BuildImageAndPush
+# Build the image and deploy in OpenShift environment
+env:
+  # Registries such as GHCR, Quay.io, and Docker Hub are supported.
+  DEVNS_IMAGE_REGISTRY: image-registry.apps.silver.devops.gov.bc.ca/c5839f-dev
+  TOOLSNS_IMAGE_REGISTRY: image-registry.apps.silver.devops.gov.bc.ca/c5839f-tools
+  IMAGE_REGISTRY_USER: ${{ secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_USER }}
+  IMAGE_REGISTRY_PASSWORD: ${{ secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_TOKEN }}
+  IMAGE_NAME: account-transfer-frontend
+  DEV_SUFFIX: -dev
+  IMAGE_TAGS:
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
-
-  # Allows you to run this workflow manually from the Actions tab
+  # Run the workflow manually, instead of on a push.
+  # The user specifies the name of the branch which is defaulted to 'main'.
+  # --------------------------------------------------------------------------
   workflow_dispatch:
+    inputs:
+      branch:
+        description: "Enter branch name:"
+        required: true
+        default: 6148-Create-Actions-for-User-Transfer-deployment
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
+  buildAndPush:
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v3
+      # Check out the code
+      - name: Check out code
+        id: checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.branch }}
+
+      # Determine the tags
+      - name: Determine image tags
+        if: env.IMAGE_TAGS == ''
+        run: |
+          echo "IMAGE_TAGS=${GITHUB_SHA::12}" | tee -a $GITHUB_ENV
+
+      # Determine the version number tag i.e. Release_1.0_<branch-name>_<build execution>
+      - name: Determine Version number
```

Comment:
we can compute the version number before tags and add version number in tag

Reply:
added version to tags